src/server/routes/groups.js

```diff
@@ -139,13 +138,10 @@ router.post('/create', async (req, res) => {
 		res.sendStatus(400);
 	} else {
 		try {
-			await getDB().tx(async t => {
-				const newGroup = new Group(undefined, req.body.name);
-				await newGroup.insert(() => t);
-				const adoptGroupsQuery = req.body.childGroups.map(gid => newGroup.adoptGroup(gid, t));
-				const adoptMetersQuery = req.body.childMeters.map(mid => newGroup.adoptMeter(mid, t));
-				return t.batch(_.flatten([adoptGroupsQuery, adoptMetersQuery]));
-			});
+			const newGroup = new Group(undefined, req.body.name);
+			await newGroup.insert();
+			await req.body.childGroups.map(gid => newGroup.adoptGroup(gid));
+			await req.body.childMeters.map(mid => newGroup.adoptMeter(mid));
```

Comment:
If these don't happen in a transaction, is it possible for the adoption of groups to succeed and the adoption of meters to fail, leaving the database in a bad state?

Reply:
You're right, the database could be in a bad state. They should be called as a transaction or a task.